### ISSUE TITLE
Exception in _remove_values_conditions()

### ISSUE THREAD

**Issue by Shrews** [ansible/ansible#30771]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible/module_utils/basic.py

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.5.0 (devel e767c7d694) last updated 2017/09/22 12:24:26 (GMT -400)
  config file = /home/shrews/.ansible.cfg
  configured module search path = ['/home/shrews/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/shrews/Devel/git/openstack-infra/zuul/.tox/py35/src/ansible/lib/ansible
  executable location = /home/shrews/Devel/git/openstack-infra/zuul/.tox/py35/bin/ansible
  python version = 3.6.2 (default, Aug 11 2017, 11:59:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]

```

##### CONFIGURATION

standard

##### OS / ENVIRONMENT

Fedora 26

##### SUMMARY

Exception thrown during module.exit_json() when one of the supplied arguments is of os.environ type.

##### STEPS TO REPRODUCE

test_module.py

```
import os

def main():
    module = AnsibleModule(
        argument_spec=dict(
            path=dict(required=False, type='str'),
        )
    )

    module.exit_json(changed=True, e=os.environ)

from ansible.module_utils.basic import *  # noqa
from ansible.module_utils.basic import AnsibleModule

if __name__ == '__main__':
    main()

```

Playbook a.yaml:

```
---
- hosts: localhost
  tasks:
     - name: Bar
       test_module:
          path: "pathname"

```

##### EXPECTED RESULTS

With Ansible 2.3:

```
$ ansible-playbook -M . a.yaml 

PLAY [localhost] *******************************************************************************

TASK [Gathering Facts] *************************************************************************
ok: [localhost]

TASK [Bar] *************************************************************************************
changed: [localhost]

PLAY RECAP *************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   

```
##### ACTUAL RESULTS

```
$ ansible-playbook -M . -vvv a.yaml 
ansible-playbook 2.5.0 (devel e767c7d694) last updated 2017/09/22 12:24:26 (GMT -400)
  config file = /home/shrews/.ansible.cfg
  configured module search path = ['/home/shrews/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/shrews/Devel/git/openstack-infra/zuul/.tox/py35/src/ansible/lib/ansible
  executable location = /home/shrews/Devel/git/openstack-infra/zuul/.tox/py35/bin/ansible-playbook
  python version = 3.6.2 (default, Aug 11 2017, 11:59:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
Using /home/shrews/.ansible.cfg as config file
Parsed /home/shrews/Devel/ansible/hosts.yaml inventory source with yaml plugin

PLAYBOOK: a.yaml ************************************************************************************************************************
1 plays in a.yaml

PLAY [localhost] ************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************
Using module file /home/shrews/Devel/git/openstack-infra/zuul/.tox/py35/src/ansible/lib/ansible/modules/system/setup.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: shrews
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241 `" && echo ansible-tmp-1506101498.485507-255357277210241="` echo /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241 `" ) && sleep 0'
<localhost> PUT /tmp/tmp_wrfp4i6 TO /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241/setup.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241/ /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241/setup.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241/setup.py; rm -rf "/home/shrews/.ansible/tmp/ansible-tmp-1506101498.485507-255357277210241/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]
META: ran handlers

TASK [Bar] ******************************************************************************************************************************
task path: /home/shrews/a.yaml:4
Using module file /home/shrews/test_module.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: shrews
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509 `" && echo ansible-tmp-1506101498.9238055-193710860524509="` echo /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509 `" ) && sleep 0'
<localhost> PUT /tmp/tmpdyoxmrf6 TO /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509/test_module.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509/ /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509/test_module.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python /home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509/test_module.py; rm -rf "/home/shrews/.ansible/tmp/ansible-tmp-1506101498.9238055-193710860524509/" > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_zVq9bq/ansible_module_test_module.py", line 16, in <module>
    main()
  File "/tmp/ansible_zVq9bq/ansible_module_test_module.py", line 10, in main
    module.exit_json(changed=True, e=os.environ)
  File "/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py", line 2263, in exit_json
  File "/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py", line 2256, in _return_formatted
  File "/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py", line 509, in remove_values
  File "/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py", line 475, in _remove_values_conditions
TypeError: instance() takes at least 1 argument (0 given)

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_zVq9bq/ansible_module_test_module.py\", line 16, in <module>\n    main()\n  File \"/tmp/ansible_zVq9bq/ansible_module_test_module.py\", line 10, in main\n    module.exit_json(changed=True, e=os.environ)\n  File \"/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2263, in exit_json\n  File \"/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2256, in _return_formatted\n  File \"/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 509, in remove_values\n  File \"/tmp/ansible_zVq9bq/ansible_modlib.zip/ansible/module_utils/basic.py\", line 475, in _remove_values_conditions\nTypeError: instance() takes at least 1 argument (0 given)\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}
	to retry, use: --limit @/home/shrews/a.retry

PLAY RECAP ******************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1   

```

**Comment by abadger** [comment]
Temporary workaround is to change module code to use ```os.environ.copy()```.  Since htis worked in 2.3.x I'll take a look at what change broke this and if there's a way to fix it for 2.4.1

**Comment by Shrews** [comment]
This has an easy fix user-side, and doesn't seem to affect anyone else for several released versions since the initial report, so going to close this.
